Dockerfile

```diff
@@ -11,7 +11,7 @@ COPY pyproject.toml /local
 ARG ANSIBLE_VER="ignore"
 
 RUN poetry config virtualenvs.create false \
-  && poetry install --no-interaction --no-ansi \
+  && poetry install --no-interaction --no-ansi --extras ansible \
```

Comment:
Please, can we do it the other way around and install ansible only if a version number is defined
instead of installing it and remove it if not required

Reply:
I'll swap the base so that it is without ansible, and add a stage of "with_ansible".

In order to support both installing the Ansible version defined in the pyproject.toml and the individually declared versions per the travis.yaml file, I believe we have to install the poetry version, then uninstall it and install the Travis declared version. There are cases in which the version number is not defined but we still want to install ansible (e.g. the ansible version in the pyproject.toml). Because of this, we can't just rely on the version keyword to determine whether or not ansible should be installed.